src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCM.java

```diff
@@ -407,224 +294,53 @@ public ListBoxModel doFillGitToolItems() {
-            }
-
-            Credentials providedCredentials = CredentialUtils.getCredentials(credentialsId);
-            if (!isBlank(credentialsId) && providedCredentials == null) {
-                return errorWithoutStack(HTTP_BAD_REQUEST, "No credentials exist for the provided credentialsId");
-            }
-
-            return bitbucketPluginConfiguration.getServerById(serverId)
-                    .map(serverConf -> {
-                        BitbucketCredentials credentials = createWithFallback(providedCredentials, serverConf);
-                        BitbucketSearchClient searchClient = bitbucketClientFactoryProvider
-                                .getClient(serverConf.getBaseUrl(), credentials)
-                                .getSearchClient(projectName);
-                        try {
-                            latestRepositories = searchClient.findRepositories(stripToEmpty(repositoryName));
-                            return okJSON(JSONObject.fromObject(latestRepositories));
-                        } catch (BitbucketClientException e) {
-                            // Something went wrong with the request to Bitbucket
-                            LOGGER.info(e.getMessage());
-                            return errorWithoutStack(HTTP_INTERNAL_ERROR, "An error occurred in Bitbucket: " + e.getMessage());
-                        }
-                    }).orElseGet(() -> errorWithoutStack(HTTP_BAD_REQUEST, "The provided Bitbucket Server serverId does not exist"));
+            return formFill.doFillRepositoryNameItems(serverId, credentialsId, projectName, repositoryName);
         }
 
+        @Override
         @POST
         public ListBoxModel doFillServerIdItems(@QueryParameter String serverId) {
             Jenkins.get().checkPermission(CONFIGURE);
-            //Filtered to only include valid server configurations
-            StandardListBoxModel model =
-                    bitbucketPluginConfiguration.getServerList()
-                            .stream()
-                            .filter(server -> server.getId().equals(serverId) ||
-                                              server.validate().kind == FormValidation.Kind.OK)
-                            .map(server ->
-                                    new Option(
-                                            server.getServerName(),
-                                            server.getId(),
-                                            server.getId().equals(serverId)))
-                            .collect(toCollection(StandardListBoxModel::new));
-            if (model.isEmpty() || model.stream().noneMatch(server -> server.value.equals(serverId))) {
-                model.includeEmptyValue();
-            }
-            return model;
+            return formFill.doFillServerIdItems(serverId);
         }
 
-        @Override
-        public String getDisplayName() {
```

Comment:
Why? I do not think we can remove this. Technically we can, but then we have ` BitbucketSCM ` as our name in the UI, which is not what we want.